composer.json

```diff
@@ -80,7 +80,9 @@
             "@composer validate --strict",
             "vendor/bin/ecs check spec src tests || true",
             "vendor/bin/psalm",
-            "vendor/bin/rector process --dry-run"
+            "vendor/bin/rector process --dry-run",
+            "tests/Application/bin/console lint:container --env dev",
+            "tests/Application/bin/console lint:container --env prod"
```

Comment:
Stumbled now over that you also have a working console the lint:container would also catch such errors. I recommend always lint for dev and prod container as they may have different set of services.